src/hotspot/share/memory/allocation.hpp

```diff
@@ -174,48 +174,137 @@ char* ReallocateHeap(char *old,
 // handles NULL pointers
 void FreeHeap(void* p);
 
-template <MEMFLAGS F> class CHeapObj {
+class CHeapObjImpl : public AllStatic {
+ public:
+   static ALWAYSINLINE void* operator new(size_t size, MEMFLAGS f) throw() {
+    return (void*)AllocateHeap(size, f);
+  }
+
+  static ALWAYSINLINE void* operator new(size_t size,
```

Comment:
Were you going to move MEMFLAGS to the second argument in the new operators also?